test/e2e/specs/test.spec.js

```diff
@@ -1,52 +1,177 @@
+
+      cy.acceptAccess();
+    });
+
+    it('should withdraw ATOMS from vault successfully', () => {
+      if (networkPhrases.isLocal) {
+        cy.visit('/');
+        cy.contains('div', /ATOM.*#5/).click();
+
+        cy.contains('div', 'Adjust Collateral')
+          .next('.grid-cols-2')
+          .within(() => {
+            cy.contains('button', /^(Deposit|Withdraw|No Action)$/).click();
+            cy.contains('button', 'Withdraw').click();
+            cy.contains('.input-label', 'Amount')
+              .next('.input-wrapper')
+              .within(() => {
+                cy.get('input[type="number"]').click();
+                cy.get('input[type="number"]').type(5);
+              });
+          });
+
+        cy.contains('button', 'Adjust Vault').click();
+        cy.confirmTransaction().then(taskCompleted => {
           expect(taskCompleted).to.be.true;
+          cy.contains('p', "Your vault's balances have been updated.", {
+            timeout: 60000,
+          }).should('exist');
+          cy.contains('Back to my vaults').click();
         });
-      });
+      }
     });
 
-    it('should adjust the collateral by performing a withdrawl and approve the transaction successfully', () => {
+    it('should create a new vault and approve the transaction successfully', () => {
       cy.visit('/');
-      cy.contains('div', /ATOM.*#5/).click();
+      if (!networkPhrases.isLocal)
+        cy.get('button').contains('Keep using Old Version').click();
+
+      if (networkPhrases.isLocal)
+        cy.contains('button', 'Add new vault').click();
+      cy.contains('button', /ATOM/).click();
+
+      cy.contains('.input-label', 'ATOM to lock up *')
+        .next()
+        .within(() => {
+          cy.get('input[type="number"]').click();
+          cy.get('input[type="number"]').clear();
+          cy.get('input[type="number"]').type(2);
+        });
+
+      cy.contains('button', 'Create Vault').click();
+
+      cy.confirmTransaction().then(taskCompleted => {
+        expect(taskCompleted).to.be.true;
+        cy.contains(
+          'p',
+          'You can manage your vaults from the "My Vaults" view.',
+          { timeout: 60000 },
+        ).should('exist');
+        cy.contains('Manage my Vaults').click();
+      });
+    });
+
+    it('should open the new vault', () => {
+      cy.get('span')
+        .contains(/My Vaults.*\(\d+\)/)
+        .children()
+        .first()
+        .spread((...element) => {
+          // Get the total number of vaults present
+          const vaultCount = Number(element[0].innerHTML.slice(2, -1));
 
+          cy.get('div.shadow-card div.text-secondary:contains("#")')
+            .should('have.length', vaultCount)
+            .spread((...vaults) => {
+              expect(
+                vaults.filter(vaultNo => !/^#\d+$/.test(vaultNo.innerHTML)),
+              ).to.be.empty;
+              // Get the vault with the largest number and click on it
```

Comment:
the largest number meaning the last one? is it also last in the list? that would be a more stable query and simpler implementation

Reply:
unfortunately no, these are sorted by the ID of the vault not by the vault #
there exists a case on local chain where the latest (and largest) vault is the third last item

Reply:
> the ID of the vault not by the vault #

By "vault #" you mean like `#2` here?

<img width="307" alt="Screenshot 2024-04-02 at 6 58 24 AM" src="https://github.com/Agoric/dapp-inter/assets/21505/8b06b027-c1a5-4e92-92cf-6b13327de2fb">

That is the unique ID of the vault in the namespace of its collateral type. The ID is a serial number so the highest ID is the most recent vault (of that collateral type).

> exists a case on local chain where the latest (and largest) vault is the third last item

Ok, so they're not sorted by increasing ID or size. I understand now that you want the latest vault, meaning the one with the largest #ID. Please clarify "largest number" because there are many numbers on a vault card.